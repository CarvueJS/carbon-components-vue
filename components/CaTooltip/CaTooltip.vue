<template>
  <div>
    <a class="bx--tooltip__trigger">
      Tooltip label
      <svg tabindex="0" data-tooltip-trigger data-tooltip-target="#unique-tooltip" width="16" height="16" viewBox="0 0 16 16" fill-rule="evenodd">
      <path d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm0 4c.6 0 1 .4 1 1s-.4 1-1 1-1-.4-1-1 .4-1 1-1zm2 8H6v-1h1V8H6V7h3v4h1v1z"></path>
    </svg>
    </a>
    <div id="unique-tooltip" data-floating-menu-direction="bottom" class="bx--tooltip">
      <p class="bx--tooltip__label">Tooltip subtitle</p>
      <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, seed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>
    </div>
  </div>
</template>

<script>
  export default {
    name: 'ca-tooltip'
  }
</script>

<style></style>
