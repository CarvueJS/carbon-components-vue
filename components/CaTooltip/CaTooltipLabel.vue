<template>
  <p class="bx--tooltip__label"><slot></slot></p>
</template>

<script>
  export default {
    name: 'ca-tooltip-label',
  };
</script>

<style></style>
