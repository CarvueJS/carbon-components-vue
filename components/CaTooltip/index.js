import CaTooltip from './CaTooltip.vue';
import CaTooltipLabel from './CaTooltipLabel.vue';
import CaTooltipTrigger from './CaTooltipTrigger.vue';

CaTooltip.install = (Vue) => {
  Vue.component(CaTooltip.name, CaTooltip);
  Vue.component(CaTooltipLabel.name, CaTooltipLabel);
  Vue.component(CaTooltipTrigger.name, CaTooltipTrigger);
};

export default CaTooltip;
