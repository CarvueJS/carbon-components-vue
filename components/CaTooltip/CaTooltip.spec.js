import { shallow } from 'vue-test-utils';
import CaTooltip from './CaTooltip.vue';

describe('CaTooltip', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaTooltip);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
})
