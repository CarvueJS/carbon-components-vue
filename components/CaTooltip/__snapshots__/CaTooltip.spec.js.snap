// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaTooltip Mathes snapshot 1`] = `<div id="unique-tooltip" data-floating-menu-direction="bottom" class="bx--tooltip"></div>`;
